src/asmcnc/skavaUI/screen_job_recovery.py

```diff
@@ -584,13 +594,18 @@ def get_info(self):
         popup_info.PopupScrollableInfo(self.sm, self.l, 760, info)
 
     def go_xy(self):
+        # Pick min out of safe z height and limit_switch_safety_distance, in case positive value is calculated, which causes errors
         z_safe_height = min(
             self.m.z_wco() + self.sm.get_screen("home").job_box.range_z[1],
             -self.m.limit_switch_safety_distance,
         )
+        # If Z is below safe height, then raise it up
         if self.m.mpos_z() < z_safe_height:
             self.m.s.write_command("G53 G0 Z%s F750" % z_safe_height)
-        self.m.s.write_command("G90 G0 X%s Y%s" % (self.pos_x, self.pos_y))
+        if self.using_inches:
+            self.m.set_machine_unit_to_inch()
+        self.m.s.write_command('G90 G0 X%s Y%s' % (self.pos_x, self.pos_y))
+        self.m.set_machine_unit_to_mm()
```

Comment:
Why do you switch back to mm if the recovered file is in inches?  Doesn't that mean, that when you re-run the job from the recover position all subsequent moves will be in mm again?

Reply:
This function is just used for the GO XY button on the job recovery screen - it sets to inches just while it moves to the position currently being displayed on that screen, then it needs to go back to mm, and when the user runs the job recovery file, then it will insert the G20 back in